lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/JobLifeCycleManager.cs

```diff
@@ -53,34 +52,46 @@ internal sealed class JobLifeCycleManager :
         {
             _cancellationDetector = cancelletionDetector;
             _timerIntervalSec = sleepIntervalSec;
+            InitTimer();
         }
 
         ~JobLifeCycleManager()
         {
             Dispose();
         }
 
-        public void OnNext(IDriverStarted value)
+        private void InitTimer()
         {
-            if (_cancellationDetector == null
-                || _cancellationDetector is JobCancellationDetectorAlwaysFalse
-                || _timerIntervalSec <= 0)
+            if (_cancellationDetector == null)
             {
-                Logger.Log(Level.Info, "Cancellation detector is null or default or timer internval is negative - no need to start Timer for job lifecycle manager. _cancellationDetector: {0}, _timer interval: {1}", _cancellationDetector, _timerIntervalSec);
+                Logger.Log(Level.Info, "Cancellation detector is null - no need to start Timer for job lifecycle manager");
                 return;
             }
 
-            Logger.Log(Level.Info, "OnDriverStart: starting timer to monitor job status. _timer interval: {0}, cancellation detector: {1}", _timerIntervalSec, _cancellationDetector);
+            if (_cancellationDetector is JobCancellationDetectorAlwaysFalse)
+            {
+                Logger.Log(Level.Info, "Cancellation detector is default - no need to start Timer for job lifecycle manager.");
+                return;
+            }
+
+            if (_timerIntervalSec <= 0)
+            {
+                Logger.Log(Level.Info, "Timer interval ({0}) is not positive - can't start Timer for job lifecycle manager.", _timerIntervalSec);
+                return;
+            }
+    
+            Logger.Log(Level.Info, "initializing timer to monitor job status. _timer interval: {0}, cancellation detector: {1}", _timerIntervalSec, _cancellationDetector);
```

Comment:
Whatever ToString implementation is. I believe default implementation prints out Type of the object, which gives us information what cancellation detector we are trying to use. I didn't use GetType() as wanted to leave some flexibility to the developer.